### PR TITLE
feat: add types, allow bullets for description

### LINKED ISSUE #23: Customisation for Description
Markdown Support or Bullet Points for Description. It helps when we have multiple things to put.

### PR DESCRIPTION
**Pull request by OmitNomis**
Add and fix some types, 

Closes #23 

Allows use of string or an array of string for desc, using an array creates bullet points. the bullets can be customized.

### This is what the config would look like: 
image

### When no `customBullet` is provided
image

### When `customBullet` is provided
image

### CODE REVIEW

File: src/data/resume-data.types.ts
Comment (co-bby) [inline src/data/resume-data.types.ts:1]: adding types is going to cause a lot of problems for people who just want to fork the project and use it. some users may not have a project, websiteurl or don't want to put out location links etc.

Comment (OmitNomis) [reply src/data/resume-data.types.ts:1]: I don't see how adding types would be a problem. 
but you're right, people might not have everything. I'll see and change the types to optional for whatever is being conditionally rendered.

File: src/app/page.tsx [hunk 1]
```diff
@@ -68,7 +69,10 @@ export default function Page() {
                   asChild
                 >
                   <a href={social.url}>
-                    <social.icon className="h-4 w-4" />
+                    {React.createElement(
+                      social.icon as React.ComponentType<{ className: string }>,
+                      { className: "h-4 w-4" },
+                    )}
```
Comment (BartoszJarocki) [inline src/app/page.tsx:75]: what's the reason for this change? what was the error?

File: src/data/resume-data.types.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { StaticImageData } from "next/image";
+export type ResumeData = {
```
Comment (BartoszJarocki) [inline src/data/resume-data.types.ts:2]: having that adds pretty much no value but adds a lot of overhead In maintaining the type. using `as const` ensures we have a type to check against and I'd rather not include this change because of that.

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 1 above)
Comment (OmitNomis) [reply src/app/page.tsx:75]: This is only if we're adding types. 
I'll remove this if types aren't necessary.

File: src/data/resume-data.types.ts (hunk 2 above)
Comment (OmitNomis) [reply src/data/resume-data.types.ts:2]: The reason I added types is because exporting as const won't allow as much customization.

For Example: 
If I have any desc with a field `customBullet` and i use it in the `page.tsx` it works fine, because the field exists. 

if I remove the field from the config, we get this error `Property 'customBullet' does not exist on type 'never'.` 

The app still works fine, but Vercel doesn't allow this. 

and for possibility of making the desc both array or string, exporting as const won't allow it unless there's atleast one array and one string in the whole config, or it'll have the type never, and the deployment fails.

**Comment by BartoszJarocki** [comment]
I suggest keeping the changes in your own fork as I see maintaining the explicit types as not needed overhead.
